src/scripts/pokemons/PokemonHelper.ts

```diff
@@ -200,13 +200,13 @@ class PokemonHelper {
         const shops = [];
         Object.entries(TownList).forEach(([townName, town]) => {
             // Check if the shop has items
-            if (town.shops && town.shops.find(s => s.items)) {
+            if (town.content.find(c => c instanceof Shop && c.items)) {
                 // If we only want to check up to a maximum region
                 const region = town.region;
                 if (maxRegion != GameConstants.Region.none && region > maxRegion) {
                     return false;
                 }
-                const hasPokemon = town.shops.find(s => s.items?.find(item => item.name == pokemonName));
+                const hasPokemon = town.content.find(c => c instanceof Shop && c.items?.find(item => item.name == pokemonName));
                 if (hasPokemon) {
```

Comment:
Might be better to instead filter only shops first so on the second find, you aren't parsing through all the content again.
Something like:
```js
const townShops = town.content.filter(c => c instanceof Shop && c.items);
if (townShops.length) {
....
                const hasPokemon = townShops.find(c => c.items.find(item => item.name == pokemonName));

Reply:
Done